Fix Modbus review comments

janiversen opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Updates for review comments from @MartinHjelmare, which was done after the code was merged.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->


```yaml
# Example configuration.yaml

modbus:
  - name: plc
    type: tcp
    host: 127.0.0.1
    port: 5020

binary_sensor:
  - platform: modbus
    scan_interval: 1
    inputs:
      - name: plc_aaa
        hub: plc
        slave: 1
        address: 0
        input_type: discrete_input

```
Remark depend a modbus tcp server.

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x ] The code change is tested and works locally.
- [ x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ x] There is no commented out code in this PR.
- [x ] I have followed the [development checklist][dev-checklist]
- [ x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

janiversen committed:
update common test for modbus integration

Remove global variable and unused constant

janiversen committed:
remove log messages from modbus setup function.

janiversen commented:
@MartinHjelmare would you please review the code, and ensure it solves your review comments, thanks in advance.

MartinHjelmare reviewed (commented):
Looks good! I still think we can drop the global:
https://github.com/home-assistant/core/pull/33447#discussion_r404021520

Is there something I'm missing about that?

janiversen committed:
Make global method local

Move simulated_read into run_tests, which allows
the holding variable to be method local, instead of
global.

janiversen commented:
No you are not missing something, I was.

I did the change, but then ran into some a pre-commit problem, and got side tracked by helping Frenck.

It is quite interesting that there are a scope difference between variables in a class and a function, I was not aware of that.

MartinHjelmare commented on tests/components/modbus/conftest.py:
```diff
@@ -46,18 +43,16 @@ def __init__(self, register_words):
-
 async def run_test(
-    hass, mock_hub, register_config, entity_domain, register_words, expected
+    hass, useMockHub, register_config, entity_domain, register_words, expected
```
Please use snake_case names for parameter names.

MartinHjelmare reviewed (commented):
Nice! Just a small convention comment.

janiversen committed:
Change parameter name to snake_case

janiversen commented:
Changed, and I also rebased on latest dev, so it is hopefully ready to get merged.

MartinHjelmare reviewed (approved):
Great!

janiversen commented:
this:
2020-04-07T07:57:49.7295319Z /__w/_temp/c2caee9c-40af-42c8-b623-1fd58fc3d1b3.sh: line 5: codecovToken: command not found
2020-04-07T07:57:49.7296032Z usage: codecov [-h] [--version] [--token TOKEN] [--file [FILE [FILE ...]]]
2020-04-07T07:57:49.7296885Z                [--flags [FLAGS [FLAGS ...]]] [--env [ENV [ENV ...]]]
2020-04-07T07:57:49.7297522Z                [--required] [--name NAME] [--gcov-root GCOV_ROOT]
2020-04-07T07:57:49.7298372Z                [--gcov-glob [GCOV_GLOB [GCOV_GLOB ...]]]
2020-04-07T07:57:49.7299006Z                [--gcov-exec GCOV_EXEC] [--gcov-args GCOV_ARGS]
2020-04-07T07:57:49.7299879Z                [-X [DISABLE [DISABLE ...]]] [--root ROOT] [--commit COMMIT]
2020-04-07T07:57:49.7300599Z                [--prefix PREFIX] [--branch BRANCH] [--build BUILD] [--pr PR]
2020-04-07T07:57:49.7302004Z                [--tag TAG] [--slug SLUG] [--url URL] [--cacert CACERT]
2020-04-07T07:57:49.7302960Z                [--dump] [-v] [--no-color]
2020-04-07T07:57:49.7303937Z codecov: error: argument --token/-t: expected one argument

can hardly by something I introduced, trying to force push again.

MartinHjelmare commented:
Yeah, we've tried to fix this this morning. Please rebase on latest dev branch and see if it works.

janiversen commented:
Done, rebase with dev as of this minute. And force push, lets how if it works

janiversen commented:
same problem! can I help to pinpoint it?

janiversen commented:
All green....yeah !!!!! Now just waiting for somebody to push the merge bottom :-)
